src/redux/auth/index.js

```diff
@@ -0,0 +1,44 @@
+import API from '../api';
+
+export const loadCurrentUser = (state) => state.auth.currentUser;
+
+const LOAD_CURRENT_USER = 'auth/current_user';
+const RESET = 'auth/reset';
+
+export const setCurrentUser = (payload) => ({
+  type: LOAD_CURRENT_USER,
+  payload,
+});
+
+export const login = (payload) => async (dispatch) => {
+  const response = await API.login(payload);
+  dispatch(setCurrentUser(response));
+};
+
+export const logout = () => async (dispatch) => {
+  await API.logout();
+  dispatch(setCurrentUser(null));
+};
```

Comment:
There is no logout function in the API so I would suggest a workaround to reset the current user from the and token to null. 
```
export const logout = () => async (dispatch) => {
  await dispatch({
    type: RESET,
  });
};
```